### PR TITLE
Add 'Quality and Contamination control' workflow

### PR DESCRIPTION
**Pull request by clsiguret**
This workflow comes from the ABRomics project and can be used for any bacterial genomic analysis:
**Quality and Contamination control for paired end data**

This workflow uses 3 databases which must be available on http://datacache.galaxyproject.org/:
- [kraken2_database](http://datacache.galaxyproject.org/managed/kraken2_databases/) (**Kraken2**): k2_minusb_20210517
- [bracken_database](http://datacache.galaxyproject.org/indexes/bracken_databases/) (**Bracken**): k2_minusb_20210517
- [ncbi_taxonomy](http://datacache.galaxyproject.org/indexes/ncbi_taxonomy/) (**Recentrifuge**): 2022-03-08

We followed the recommendation to publish the workflow (licence, author list and email, steps annotations...)
- [x] Workflow imported from [usegalaxy.fr](https://usegalaxy.fr/) using planemo `workflow_test_init` to use a valid invocation
- [x] Workflow tests passed using `planemo test` command on the [usegalaxy.eu](https://usegalaxy.eu/) server
- [x] Readme.md written
- [x] CHANGELOG.md written
- [x] .dockstore.yml written

### CODE REVIEW

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+        },
+        {
+            "alternateName": "clsiguret",
+            "class": "Person",
+            "identifier": "https://orcid.org/0009-0005-6140-0379",
+            "name": "Clea Siguret"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "GPL-3.0-or-later",
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:30]:
```suggestion
    "release": "0.1",
    "license": "GPL-3.0-or-later",
```
This will determine the version of the github release

File: workflows/bacterial_genomics/quality_and_contamination_control/test-data/paired_r1.fastq.gz
[the remark this answers is not in the document]
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/test-data/paired_r1.fastq.gz:1]: Done. Thanks !

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga [hunk 2]
```diff
@@ -0,0 +1,811 @@
...
+            "tool_id": null,
+            "tool_state": "{\"optional\": false, \"format\": [\"fastq\", \"fastq.gz\", \"fastqsanger\", \"fastqsanger.gz\"], \"tag\": \"\"}",
+            "tool_version": null,
+            "type": "data_input",
+            "uuid": "0f8fa828-7694-4a4f-909c-3d2d468013af",
+            "when": null,
+            "workflow_outputs": []
+        },
+        "2": {
+            "annotation": "Select the Kraken2 database used to the taxonomic assignation, the same database type need to be used by Bracken",
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:89]: Since these are the same values I would just use a single parameter. This way it is also impossible for users to select conflicting values.

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga [hunk 3]
```diff
@@ -0,0 +1,811 @@
...
+            "tool_version": null,
+            "type": "parameter_input",
+            "uuid": "83589940-64a6-4b37-9140-9684ab565a5b",
+            "when": null,
+            "workflow_outputs": []
+        },
+        "3": {
+            "annotation": "Select the Bracken database used to the taxonomic assignation, the same database type need to be used by Kraken2",
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:116]:
```suggestion
            "annotation": "Select the database to use for assigning taxonomies.",
```

File: workflows/bacterial_genomics/quality_and_contamination_control/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+## [0.1] - 04-06-2024
```
Comment (wm75) [inline workflows/bacterial_genomics/quality_and_contamination_control/CHANGELOG.md:3]: doesn't match release (1.0) stated in WF file

File: workflows/bacterial_genomics/quality_and_contamination_control/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Quality and Contamination control subworkflow for paired end data (v1.0)
+
+This subworkflow use paired-end illumina fastq(.gz) files and executes the following steps:
```
Comment (wm75) [inline workflows/bacterial_genomics/quality_and_contamination_control/README.md:3]:
```suggestion
This subworkflow uses paired-end illumina fastq(.gz) files and executes the following steps:
```

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga [hunk 6]
```diff
@@ -0,0 +1,811 @@
...
+                {
+                    "label": "tooldistillator_results",
+                    "output_name": "output_json",
+                    "uuid": "cc18f393-547e-4d7f-b836-fedda6aa613f"
+                }
+            ]
+        },
+        "10": {
+            "annotation": "ToolDistillator summarize groups all JSON file into a unique JSON file",
+            "content_id": "toolshed.g2.bx.psu.edu/repos/iuc/tooldistillator_summarize/tooldistillator_summarize/0.8.5.0+galaxy0",
```
Comment (wm75) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:741]: same naive question as in #438 - why is this necessary?

Comment (wm75) [reply workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:741]: ah, got it now at last - it's one tool run of ToolDistillator, but that produces a collection of json files that you are merging at this step.

File: workflows/bacterial_genomics/quality_and_contamination_control/README.md (hunk 5 above)
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/README.md]: Thanks for the correction  :flushed:

File: workflows/bacterial_genomics/quality_and_contamination_control/CHANGELOG.md (hunk 4 above)
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/CHANGELOG.md]: Thanks !

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga (hunk 2 above)
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga]: Thanks, i didn't think about it  :flushed:

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga (hunk 3 above)
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga]: Thanks !

**Comment by bebatut** [comment]
We need this PR to be merged as soon as possible to have time to update our workflows and tutorials before GCC and the French Bioinformatics conference next week.
If I do not get any "No do not merge" by the middle of the day, I will merge it and we will make a V2 of the workflow later.

File: workflows/bacterial_genomics/quality_and_contamination_control/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Quality and Contamination control workflow for paired end data (v1.0)
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/README.md:1]: Is this a general purpose workflow or is any part of this tailored to bacterial genome assembly ?

Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/README.md:1]: The workflow has a general purpose. Simply use the trimmed fastqs from this workflow to launch “bacterial genome assembly”

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga [hunk 8]
```diff
@@ -0,0 +1,786 @@
...
+            "class": "Person",
+            "identifier": "https://orcid.org/0009-0005-6140-0379",
+            "name": "Clea Siguret",
+            "alternateName": "clsiguret"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "GPL-3.0-or-later",
+    "release": "1.0",
+    "name": "quality_and_contamination_control",
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:33]:
```suggestion
    "name": "Quality and Contamination Control For Genome Assembly",
```
Does that sound good to you ?

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml [hunk 9]
```diff
@@ -0,0 +1,63 @@
+- doc: Test outline for quality_and_contamination_control.ga
+  job:
+    Input sequence reads (forward):
+      class: File
+      path: https://zenodo.org/records/11484215/files/paired_r1.fastq.gz
+      filetype: fastq.gz
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml:6]:
```suggestion
      filetype: fastqsanger.gz
```

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    Input sequence reads (reverse):
+      class: File
+      path: https://zenodo.org/records/11484215/files/paired_r2.fastq.gz
+      filetype: fastq.gz
```
Comment (mvdbeek) [inline workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml:10]:
```suggestion
      filetype: fastqsanger.gz
```

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga (hunk 8 above)
Comment (clsiguret) [reply workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control.ga:33]: Yes, thanks :smile:

File: workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml (hunk 10 above)
Comment (mvdbeek) [reply workflows/bacterial_genomics/quality_and_contamination_control/quality_and_contamination_control-tests.yml:10]: It's not that important, but many (other) tools only accept inputs in sanger encoding, and everything is sanger encoded these days.

**Review by mvdbeek — APPROVED** [review]
Again, very nice work, thanks a lot!

**Comment by clsiguret** [comment]
> Again, very nice work, thanks a lot!

Thanks you (@mvdbeek) and @wm75 :smile:

**Comment by mvdbeek** [comment]
Attention: deployment failure!

https://github.com/galaxyproject/iwc/actions/runs/9544770124

**Comment by clsiguret** [comment]
> Attention: deployment failure!
> 
> https://github.com/galaxyproject/iwc/actions/runs/9544770124

the problem comes from `workflows/bacterial_genomics/bacterial_genome_assembly` ?

**Comment by mvdbeek** [comment]
No, the version in the .dockstore.yml field is the version of the .dockstore.yml schema, it should be 1.2 always, I've fixed that in https://github.com/galaxyproject/iwc/commit/9458220b6fd44bb797a645877676cdd9ba7f2233
